examples/jsm/nodes/materials/nodes/BasicNode.js

```diff
@@ -21,22 +21,45 @@ BasicNode.prototype.generate = function ( builder ) {
 
 		var position = this.position ? this.position.analyzeAndFlow( builder, 'v3', { cache: 'position' } ) : undefined;
 
+		builder.addParsCode( [
+			"varying vec3 vViewPosition;",
+
+			"#ifndef FLAT_SHADED",
+
+			" varying vec3 vNormal;",
+
+			"#endif",
+		].join( "\n" ) );
+
 		var output = [
-			"vec3 transformed = position;"
+			"#include <beginnormal_vertex>",
+			"#include <defaultnormal_vertex>",
+
+			"#ifndef FLAT_SHADED", // Normal computed with derivatives when FLAT_SHADED
+
+			" vNormal = normalize( transformedNormal );",
+
+			"#endif",
+
+			"#include <begin_vertex>",
 		];
 
 		if ( position ) {
 
 			output.push(
 				position.code,
-				position.result ? "gl_Position = projectionMatrix * modelViewMatrix * vec4(" + position.result + ", 1.0);" : ''
+				position.result ? "transformed = " + position.result + ";" : ''
 			);
 
-		} else {
+		}
 
-			output.push( "gl_Position = projectionMatrix * modelViewMatrix * vec4(transformed, 1.0);" );
+		output.push(
+			"#include <project_vertex>",
```

Comment:
I think that we can add all systems includes like StandardNodeMaterial?
```
"#include <morphtarget_vertex>",
"#include <skinning_vertex>",
"#include <project_vertex>",
"#include <fog_vertex>",
"#include <logdepthbuf_vertex>",
"#include <clipping_planes_vertex>",

"	vViewPosition = - mvPosition.xyz;",

"#include <worldpos_vertex>",
"#include <shadowmap_vertex>"
```

Reply:
What would actually be the limit line between the `BasicNodeMaterial` and the `StandardNodeMaterial`? I had in mind that the `BasicNodeMaterial` would be very similar to the `ShaderMaterial` class in ThreeJS, as it would include very small amount of logic by default and require the user to implement almost all the shaders logic.

Reply:
I think that is necessary add system includes(global settings)... `BasicNodeMaterial` is more optimized, it must not include PBR calcs.

Reply:
I've not forgotten about this. I will finish this PR soon-ish :)

Reply:
Done :)